Update documentation

ABaldwinHunter opened the pull request:
- Add debug section
- Update analyze command flags
@codeclimate/review

pbrisbin commented on README.md:
```diff
@@ -90,7 +90,8 @@ Available commands:
 
-* `analyze`: Analyze all relevant files in the current working directory. All engines that are enabled in your `.codeclimate.yml` file will run, one after another. The `-f` (or `format`) argument allows you to set the output format of the analysis (using `json`, `text`, or `html`).
+* `analyze`: Analyze all relevant files in the current working directory. All engines that are enabled in your `.codeclimate.yml` file will run, one after another. The `-f` (or `format`) argument allows you to set the output format of the analysis (using `json`, `text`, or `html`). The `--dev` flag lets you run your own engines during
+development.
```
I think `--dev` is under-explained here. I would either explain it more (i.e. "lets you run engines not known to the CLI, for example if you're engine author developing your own, unreleased image"), or not mention it at all.

pbrisbin commented on README.md:
```diff
@@ -102,6 +103,33 @@ The following is a brief explanation of each available command.
 * `validate-config`: Validates the `.codeclimate.yml` file in the current working directory.
 * `version`: Displays the current version of the Code Climate CLI.
 
+## Debugging
```
I would call this section something else (maybe Environment Variables?). Only `CODECLIMATE_DEBUG` has to do with debugging, the other two don't.

pbrisbin commented on README.md:
````diff
@@ -102,6 +103,33 @@ The following is a brief explanation of each available command.
+- To increase the amount of time each engine container may run (default 15 min):
+
+  ```
+  CONTAINER_TIMEOUT_SECONDS=1800 codeclimate analyze // 30 minutes
````
It makes me sad these aren't `CODECLIMATE_` prefixed. Nothing to address now obviously, but just flagging it as a bit of debt.

ABaldwinHunter commented:
@pbrisbin Ready for another look!

- added additional information about `--dev`
- renamed the new section to `Environment Variables` (should we remove the intro about using
items in this section for debugging?)
- enabled markdownlint and fixed related issues. (sorry - makes diff a little hard to read)

pbrisbin commented on DEVELOPERS.md:
```diff
@@ -1,4 +1,6 @@
-Information for Code Climate CLI developers and contributors.
+# Development
```
WDYT about disabling this check. All three cases where you listened to it feel worse to me.

ABaldwinHunter replied:
okay

pbrisbin commented:
> should we remove the intro about using items in this section for debugging?

IMO, yes.

LGTM pending that change.
